### PR TITLE
Added pkg-config file and .gitignore

### PR DESCRIPTION
**Pull request by oxoocoffee**
I was trying to build mongo-c-driver 1.7.0 on Mac OS X and it was failing to detect since missing pc file

### CODE REVIEW

**Comment by oxoocoffee** [comment]
I signed it!

File: cmake/snappy.pc.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_PREFIX@/lib
```
Comment (junghans) [inline cmake/snappy.pc.in]: I would really use `include(GNUInstallDirs)` and not hardcode `lib`.

**Comment by cmumford** [comment]
@googlebot rescan

**Comment by lavoiesl** [comment]
Any update on this PR?

I’m unable to compile the snappy ruby gem on nix because it can't detect the correct paths. This would be fixed by having a pkg config file.

**Comment by pwnall** [comment]
We will not add pkg-config support for now.

We don't want to get into supporting individual package managers, because the C++ ecosystem hasn't converged into a single solution, and there are quite a few equally reasonable candidates. We can't justify the investment into supporting all the package managers, or into supporting a couple of package managers and trying to rationalize where we drew the line.

**Comment by lovesegfault** [comment]
@pwnall By now pkg-config is a crucial part of the C/++ ecosystem, in particular for developers downstream trying to build bindings or link snappy into their non-C/++ projects.

I ended up on this issue because I need to link snappy into a Rust binary and have to write special code _only_ for snappy because it doesn't provide a pc file.

Given that the complexity cost of this change is very small (24 lines) I ask you to please reconsider this.

**Pull request closed without merging**